packages/agoric-cli/src/commands/ec.js

```diff
@@ -1,126 +1,304 @@
+export const makeEconomicCommiteeCommand = async (_logger, io = {}) => {
+  const {
+    // Allow caller to provide access explicitly, but
+    // default to conventional ambient IO facilities.
+    env = process.env,
+    stdout = process.stdout,
+    fetch = globalThis.fetch,
+    execFileSync = execFileSyncAmbient,
+    delay = ms => new Promise(resolve => setTimeout(resolve, ms)),
+  } = io;
+
   const ec = new Command('ec').description('Economic Committee commands');
 
+  /** @param {string} literalOrName */
+  const normalizeAddress = literalOrName =>
+    normalizeAddressWithOptions(literalOrName, { keyringBackend: 'test' });
+
+  const show = (info, indent) =>
+    stdout.write(`${JSON.stringify(info, null, indent ? 2 : undefined)}\n`);
+
+  const findContinuingIds = (current, agoricNames) => {
+    // XXX should runtime type-check
+    /** @type {{ offerToUsedInvitation: [string, Amount<'set'>][]}} */
+    const { offerToUsedInvitation: entries } = /** @type {any} */ (current);
+
+    assert(Array.isArray(entries));
+
+    const keyOf = (obj, val) => {
+      const found = Object.entries(obj).find(e => e[1] === val);
+      return found && found[0];
+    };
+
+    const found = [];
+    for (const [offerId, { value }] of entries) {
+      /** @type {{ description: string, instance: unknown }[]} */
+      const [{ description, instance }] = value;
+      if (
+        description === 'charter member invitation' ||
+        /Voter\d+/.test(description)
+      ) {
+        const instanceName = keyOf(agoricNames.instance, instance);
+        found.push({ instance, instanceName, description, offerId });
+      }
+    }
+    return found;
+  };
+
+  const abortIfSeen = (instanceName, found) => {
+    const done = found.filter(it => it.instanceName === instanceName);
+    if (done.length > 0) {
+      console.warn(`invitation to ${instanceName} already accepted`, done);
+      throw new CommanderError(1, 'ELREADY', `already accepted`);
+    }
+  };
+
+  /**
+   * Sign and send an offer, given a sendFrom address;
+   * else print it.
+   *
+   * Given sendFrom and instanceName, abort early if
+   * such an offer is already accepted.
+   *
+   * @param {{
+   *   toOffer: (agoricNames: *, current: *) => OfferSpec,
+   *   sendFrom: string,
+   *   instanceName?: string,
+   * }} detail
+   * @param {Awaited<ReturnType<makeRpcUtils>>} [optUtils]
+   */
+  const processOffer = async function (
+    { toOffer, sendFrom, instanceName },
+    optUtils,
+  ) {
+    const networkConfig = await getNetworkConfig(env);
+    const utils = await (optUtils || makeRpcUtils({ fetch }));
+    const { agoricNames, readLatestHead } = utils;
+
+    let current;
+    if (sendFrom) {
+      current = await readLatestHead(`published.wallet.${sendFrom}.current`);
+    }
+
+    const offer = toOffer(agoricNames, current);
+    if (!sendFrom) {
+      outputExecuteOfferAction(offer);
+
+      console.warn('Now execute the prepared offer');
```

Comment:
When would we want to `outputExecuteOfferAction` without this user instruction? If not then please put into the fn